Src/tim.c

```diff
@@ -53,9 +53,13 @@ void MX_TIM16_Init(void)
   TIM_BreakDeadTimeConfigTypeDef sBreakDeadTimeConfig;
 
   htim16.Instance = TIM16;
-  htim16.Init.Prescaler = 0;
+
+  
+  // 16bit timer is ~60000 counts. Since desired period is 20ms => 60000/0.02 = 3MHz. 
+  // Since clock is 48MHz, prescaler should be 16. 
+  htim16.Init.Prescaler = 16;
   htim16.Init.CounterMode = TIM_COUNTERMODE_UP;
-  htim16.Init.Period = 0;
+  htim16.Init.Period = 0.02; // For a frequency of 50hz, period is 20ms.  
```

Comment:
1. Saying a 16-bit timer is around 60,000 counts will lead to a lot of inaccuracies in your calculation of a prescalar and period. I would revisit this and use 2^16-1 in your calculations. 
2. The period is actually measured in counts (a number between 0 and 2^16-1), not seconds. The period value is the maximum value that the timer will count till before restarting from 0. (ex. if I set the period to 30000, the timer will count to 30000 and then go back to 0). 

Tip: Choose a prescalar such that counting from 0 to 2^16-1 brings you as close to 20 ms as possible (but make sure it's greater than 20 ms [ex. 21 ms]). Let's say the prescalar you choose gives you a period of 25 ms. That means, it will take your timer 25 ms to count from 0-2^16-1. Now, you can modify the Period property of the timer to be (2^16-1)/2. What this will do is make the period (time between consecutive 0s in the timer) of the timer 12.5 ms.